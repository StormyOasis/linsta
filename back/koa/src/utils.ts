import jwt from "jsonwebtoken";
import { Context } from "koa";

import {
  config,
  DBConnector,
  EDGE_POST_TO_COMMENT,
  EDGE_USER_FOLLOWS,
  EDGE_USER_LIKED_POST,
  getPfpByUserId,
  logger,
  type Like,
  type Post,
  handleSuccess as sharedHandleSuccess,
  handleValidationError as sharedHandleValidationError
} from "@linsta/shared";

export const isUserAuthorized = (ctx: Context, userId: string): boolean => {
    return ctx.state.user?.id === userId;
}

export const handleSuccess = (ctx: Context, result: unknown) => {
    const data = sharedHandleSuccess(result);

    ctx.status = data.statusCode;
    ctx.body = data.body;
    ctx.set(data.headers);
}

export const handleValidationError = (ctx: Context, error: string, statusCode: number = 400) => {
    const data = sharedHandleValidationError(error, statusCode);

    ctx.status = data.statusCode;
    ctx.body = data.body;
    ctx.set(data.headers);
}

export type JWTData = {
    id: string;
}

export const verifyJWT = async (ctx: Context, next: () => unknown) => {
    const authHeader: string | undefined = ctx.request.headers['authorization'];
    let token: string | undefined = undefined;
    if (authHeader) {
        const bearer = authHeader.split(' ');
        token = bearer[1];
    }

    if (!token) {
        ctx.res.statusCode = 403;
        ctx.body = { status: "Invalid token" };
        return;
    }

    let result = null;
    await jwt.verify(
        token as string,
        config.auth.jwt.secret,
        (err, decoded) => {
            const data = decoded as JWTData;
            if (err || decoded == null) {
                ctx.res.statusCode = 403;
                ctx.body = { status: "Invalid token" };
                return;
            }
            result = data.id;
        })

    if (result == null) {
        ctx.res.statusCode = 403;
        ctx.body = { status: "Error verifying id" };
        return;
    }

    ctx.res.statusCode = 200;
    ctx.body = { status: "OK" };

    await next();
}

export const addPfpsToPosts = async (posts: Record<string, Post>) => {
    await Promise.all(Object.values(posts).map(async post => {
        post.user.pfp = await getPfpByUserId(false, post.user.userId);
    }));
}

export const addCommentCountsToPosts = async (posts: Record<string, Post>, postIds: string[]) => {
    const __ = DBConnector.__();
    const commentResults = await (await DBConnector.getGraph()).V(postIds)
        .project("postId", "commentCount")
        .by(__.id())
        .by(__.outE(EDGE_POST_TO_COMMENT).count())
        .toList();

    for (const result of commentResults) {
        const data = DBConnector.unwrapResult(result);
        const parsed = DBConnector.parseGraphResult<{ postId: string; commentCount: number }>(
            data,
            ["postId", "commentCount"]
        );

        if (parsed.postId in posts) {
            posts[parsed.postId].global.commentCount = parsed.commentCount;
        }
    }
};

export const addLikesToPosts = async (posts: Record<string, Post>, postIds: string[]) => {
    const __ = DBConnector.__();
    const likeResults = await (await DBConnector.getGraph()).V(postIds)
        .filter(__.inE(EDGE_USER_LIKED_POST).count().is(DBConnector.P().gt(0)))
        .project("postId", "users")
        .by(__.id())
        .by(__.inE(EDGE_USER_LIKED_POST)
            .outV()
            .project('profileId', 'userName', 'pfp', 'firstName', 'lastName', 'userId')
            .by("profileId")
            .by("userName")
            .by("pfp")
            .by("firstName")
            .by("lastName")
            .by(__.id())
            .fold())
        .toList();

    for (const result of likeResults) {
        const data = DBConnector.unwrapResult(result);
        const parsed = DBConnector.parseGraphResult<{ postId: string; users: unknown[] }>(data, ["postId", "users"]);

        const { postId, users } = parsed;

        if (!posts[postId]) {
            continue;
        }

        const post = posts[postId];
        post.global.likes = [];

        for (const user of users ?? []) {
            const userData = DBConnector.unwrapResult(user);
            const userParsed = DBConnector.parseGraphResult<Like>(userData, [
                "userName",
                "userId",
                "profileId",
                "firstName",
                "lastName",
                "pfp"
            ]);

            post.global.likes.push({
                userName: userParsed.userName,
                userId: userParsed.userId,
                profileId: userParsed.profileId,
                firstName: userParsed.firstName,
                lastName: userParsed.lastName,
                pfp: userParsed.pfp
            });
        }
    }
};

export const getFollowingUserIds = async (userId: string): Promise<string[]> => {
    try {
        const results = await (await DBConnector.getGraph())
            .V(userId)
            .out(EDGE_USER_FOLLOWS)
            .toList();

        return results.map(result => {
            const vertex = DBConnector.unwrapResult(result);
            if (vertex instanceof Map) {
                return String(vertex.get("id"));
            } else if (typeof vertex === "object" && vertex !== null && "id" in vertex) {
                return String((vertex as { id: unknown }).id);
            }
            throw new Error("Unexpected vertex format");
        });

    } catch (err) {
        logger.error("Failed to get following user IDs", { userId, error: err });
        return [];
    }
}