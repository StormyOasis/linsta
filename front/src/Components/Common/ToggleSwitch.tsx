import React, { useState } from "react";
import { styled } from "styled-components";

const SwitchLabel = styled.label`
  display: flex;
  position: relative;
  justify-content: space-between;
  background: ${props => props.theme['colors'].borderDarkColor};
  align-items: center;  
  width: 64px;
  height: 24px;    
  cursor: pointer;  
  border-radius: 50px;  
  transition: background-color 0.2s;

  &:active span {
    width: 40px;
  }
`;

const SwitchInput = styled.input`
  height: 0;
  width: 0;
  visibility: hidden;

  &:checked + label {
    background: ${props => props.theme['colors'].buttonDefaultColor};
  }

  &:checked + label span {
    left: calc(100% - 2px);
    transform: translateX(-100%);    
    background: ${props => props.theme['colors'].backgroundColor};
  }
`;

const SwitchSliderSpan = styled.span`
  content: '';
  position: absolute;
  top: 2px;
  left: 2px;
  width: 20px;
  height: 20px;
  border-radius: 20px;
  transition: 0.2s;
  background: ${props => props.theme['colors'].backgroundColor};
`;

type ToggleSwitchProps = {
    isChecked?: boolean;
    onChange?: (isChecked: boolean) => void;
}

const ToggleSwitch: React.FC<ToggleSwitchProps> = (props: ToggleSwitchProps) => {
    const [isChecked, setIsChecked] = useState(props.isChecked || false);
     //since it's an id, make sure it's unique in case there's multiple toggles on same page
    const [id, _setId] = useState(crypto.randomUUID());

    const handleChange = (_event: any) => {
        const checked = !isChecked;
        setIsChecked(checked);
        if(props.onChange) {
            props?.onChange(checked);
        }        
    }

    return (
        <>
            <SwitchInput id={id} type="checkbox" onChange={handleChange} checked={isChecked}></SwitchInput>
            <SwitchLabel htmlFor={id}>            
                <SwitchSliderSpan>
                </SwitchSliderSpan>
            </SwitchLabel>            
        </>
    );
};

export default ToggleSwitch;