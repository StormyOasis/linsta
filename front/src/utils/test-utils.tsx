import React, { ReactElement } from 'react';
import { render, RenderOptions } from '@testing-library/react';
import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router-dom';
import { RootState } from "../Components/Redux/redux";
import { authInitialState } from '../Components/Redux/slices/auth.slice';
import { modalInitialState } from '../Components/Redux/slices/modals.slice';
import { profileInitialState } from '../Components/Redux/slices/profile.slice';
import { miscInitialState } from '../Components/Redux/slices/misc.slice';
import { configureStore } from '@reduxjs/toolkit';

import Theme from '../Components/Themes/Theme';

export const mockInitialState = (overrides?: Partial<RootState>): RootState => ({
    auth: { ...authInitialState, ...(overrides?.auth ?? {}) },
    modal: { ...modalInitialState, ...(overrides?.modal ?? {}) },
    profile: { ...profileInitialState, ...(overrides?.profile ?? {}) },
    misc: { ...miscInitialState, ...(overrides?.misc ?? {}) },
});

type ExtendedRenderOptions = {
    store?: ReturnType<typeof configureStore>;
    preloadedState?: any;
    route?: string; // ‚Üê allow overriding the initial route
} & Omit<RenderOptions, "queries">;

const dummyAuthReducer = (state = { user: null, isLoading: false, error: null }, _action: any) => state;
const dummyModalReducer = (state = { isOpen: false }, _action: any) => state;
const dummyProfileReducer = (state = { name: '', email: '' }, _action: any) => state;
const dummyMiscReducer = (state = { someData: null }, _action: any) => state;

const renderWithStore = (
    ui: ReactElement,
    {
        preloadedState,
        store = configureStore({
            reducer: {
                auth: dummyAuthReducer,
                modal: dummyModalReducer,
                profile: dummyProfileReducer,
                misc: dummyMiscReducer,
            },
            preloadedState,
        }),
        route = "/",
        ...renderOptions
    }: ExtendedRenderOptions = {}
) => {
    const Wrapper: React.FC<{ children: React.ReactNode }> = ({ children }) => {
        return (
            <Provider store={store}>
                <MemoryRouter initialEntries={[route]}>
                    <Theme>{children}</Theme>
                </MemoryRouter>
            </Provider>
        );
    };

    return {
        ...render(ui, { wrapper: Wrapper, ...renderOptions }),
        store,
    };
};

export * from "@testing-library/react";
export { renderWithStore };
export type { ExtendedRenderOptions };