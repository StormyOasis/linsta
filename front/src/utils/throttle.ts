import { useRef } from 'react';

const useThrottle = (callback: Function, delay: number) => {
    const lastCall = useRef(0);

    return (...args: any[]) => {
        const now = new Date().getTime();
        if (now - lastCall.current >= delay) {
            lastCall.current = now;
            callback(...args);
        }
    };
};

export default useThrottle;