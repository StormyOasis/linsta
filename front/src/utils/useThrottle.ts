import { useRef, useCallback } from 'react';

/**
 * React hook for throttling a callback function.
 * The callback will only be invoked at most once every `delay` milliseconds.
 *
 * @param callback - The function to throttle.
 * @param delay - The minimum delay (ms) between calls.
 * @returns A throttled version of the callback.
 */
const useThrottle = <T extends (...args: any[]) => void>(callback: T, delay: number) => {
    const lastCall = useRef<number>(0);

    return useCallback(
        (...args: Parameters<T>) => {
            const now = performance.now();
            if (now - lastCall.current >= delay) {
                lastCall.current = now;
                callback(...args);
            }
        },
        [callback, delay]
    );
};

export default useThrottle;