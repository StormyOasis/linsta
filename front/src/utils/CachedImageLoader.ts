import { clear, del, get, set } from 'idb-keyval';
import { blobToBase64 } from "./utils";

export const loadImageCached = async (id: string, url: string): Promise<string> => {
    // First try to get the image from the cache
    const value = await get(id);

    if(value === undefined) {
        // Image not in the cache so have to fetch it
        try {
            const result = await fetch(url);
            if(result.status === 200) {
                const blob = await result.blob();
                const b64:string = await blobToBase64(blob) as string;

                // add the base64 data into the cache and return it
                await set(id, b64);

                return b64;
            }
        } catch(err) {
            console.log(err);
            return url; //default to what was passed in
        }

    } else {
        // Image is in the cache already as a base64 string, use that as is
        return value;
    }

    return url; //default to what was passed in
}

export const setImage = async (id: string, data: string) => {     
    await set(id, data);
}

export const deleteFromCache = async (id: string) => {
    // First try to get the image from the cache
    const value = await get(id);

    if(value === undefined) {
        // key not present in cache so do nothing
        return;
    }

    // delete from the cache
    await del(id);
}

export const clearCache = async () => {
    await clear();
}