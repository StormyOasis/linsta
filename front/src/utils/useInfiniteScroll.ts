import React, { useEffect, useRef } from 'react';
import useThrottle from './useThrottle';

/**
 * Type for the async load more function.
 */
type AsyncLoadMoreFunction = () => Promise<void>;

/**
 * Custom hook for infinite scrolling.
 * Attaches a throttled scroll event listener to a container and triggers `loadMore`
 * when the user scrolls near the bottom.
 *
 * @param loadMore - Async function to load more items.
 * @param isLoading - Whether a load is currently in progress.
 * @param containerRef - Ref to the scrollable container element.
 */
const useInfiniteScroll = (loadMore: AsyncLoadMoreFunction, 
    isLoading: boolean, 
    childRef: React.MutableRefObject<HTMLDivElement | null>) => {
        
    const hasScrolled = useRef<boolean>(false);  // Track if the user has scrolled

    // Throttled scroll handler
    const throttledHandleScroll = useThrottle(async () => {
        if (typeof window !== 'undefined' && childRef.current) {
            const element = childRef.current as HTMLElement;
            const currentScroll = window.innerHeight + element.scrollTop;

            if (currentScroll + 256 >= element.scrollHeight && !isLoading && !hasScrolled.current) {
                hasScrolled.current = true;  // Mark as scrolled at least once
                await loadMore();
            }
        }
    }, 200);

    useEffect(() => {
        const childElement = childRef.current;
        if (childElement) {
            childElement.addEventListener('scroll', throttledHandleScroll);
        }

        // Clean up event listener on unmount
        return () => {
            if (childElement) {
                childElement.removeEventListener('scroll', throttledHandleScroll);
            }
        };
    }, [throttledHandleScroll, childRef]);

    // Reset trigger flag when loading finishes
    useEffect(() => {
        if (!isLoading) {
            hasScrolled.current = false;
        }
    }, [isLoading]);
}

export default useInfiniteScroll;