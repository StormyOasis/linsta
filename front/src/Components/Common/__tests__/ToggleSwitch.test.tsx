import React from "react";
import { screen, fireEvent } from "@testing-library/react";
import ToggleSwitch from "../ToggleSwitch";
import '@testing-library/jest-dom';
import { renderWithStore } from '../../../utils/test-utils';

// Polyfill for crypto.randomUUID in Jest/node
if (!global.crypto) {
    global.crypto = {} as Crypto;
}
if (!global.crypto.randomUUID) {
    global.crypto.randomUUID = () => "test-uuid-1234-5678-910";
}

describe("ToggleSwitch", () => {
    it("renders the switch input and label", () => {
        const r = renderWithStore(<ToggleSwitch />);
        const input = screen.getByTestId("toggle-checkbox");
        expect(input).toBeInTheDocument();
        expect(r).toMatchSnapshot();
    });

    it("is unchecked by default", () => {
        const r = renderWithStore(<ToggleSwitch />);
        const input = screen.getByTestId("toggle-checkbox");
        expect(input).not.toBeChecked();
        expect(r).toMatchSnapshot();
    });

    it("respects isChecked prop", () => {
        const r = renderWithStore(<ToggleSwitch isChecked={true} />);
        const input = screen.getByTestId("toggle-checkbox");
        expect(input).toBeChecked();
        expect(r).toMatchSnapshot();
    });

    it("calls onChange with correct value when toggled", () => {
        const handleChange = jest.fn();
        const r = renderWithStore(<ToggleSwitch onChange={handleChange} />);
        const input = screen.getByTestId("toggle-checkbox");
        fireEvent.click(input);
        expect(handleChange).toHaveBeenCalledWith(true);
        fireEvent.click(input);
        expect(handleChange).toHaveBeenCalledWith(false);
        expect(r).toMatchSnapshot();
    });

    it("toggles checked state on click", () => {
        const r = renderWithStore(<ToggleSwitch />);
        const input = screen.getByTestId("toggle-checkbox");
        expect(input).not.toBeChecked();
        fireEvent.click(input);
        expect(input).toBeChecked();
        fireEvent.click(input);
        expect(input).not.toBeChecked();
        expect(r).toMatchSnapshot();
    });
});