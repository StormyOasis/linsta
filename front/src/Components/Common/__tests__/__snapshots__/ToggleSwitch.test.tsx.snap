// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ToggleSwitch calls onChange with correct value when toggled 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <input
        class="ToggleSwitch__SwitchInput-sc-1n3fzwh-1 Fwlpj"
        data-testid="toggle-checkbox"
        id="test-uuid-1234-5678-910"
        type="checkbox"
      />
      <label
        class="ToggleSwitch__SwitchLabel-sc-1n3fzwh-0 hFRJiv"
        for="test-uuid-1234-5678-910"
      >
        <span
          class="CombinedStyling__Span-sc-c3jdxk-1 ToggleSwitch__SwitchSliderSpan-sc-1n3fzwh-2 kSRMWv hLssYN"
        />
      </label>
    </div>
  </body>,
  "container": <div>
    <input
      class="ToggleSwitch__SwitchInput-sc-1n3fzwh-1 Fwlpj"
      data-testid="toggle-checkbox"
      id="test-uuid-1234-5678-910"
      type="checkbox"
    />
    <label
      class="ToggleSwitch__SwitchLabel-sc-1n3fzwh-0 hFRJiv"
      for="test-uuid-1234-5678-910"
    >
      <span
        class="CombinedStyling__Span-sc-c3jdxk-1 ToggleSwitch__SwitchSliderSpan-sc-1n3fzwh-2 kSRMWv hLssYN"
      />
    </label>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "store": {
    "@@observable": [Function],
    "dispatch": [Function],
    "getState": [Function],
    "replaceReducer": [Function],
    "subscribe": [Function],
  },
  "unmount": [Function],
}
`;

exports[`ToggleSwitch is unchecked by default 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <input
        class="ToggleSwitch__SwitchInput-sc-1n3fzwh-1 Fwlpj"
        data-testid="toggle-checkbox"
        id="test-uuid-1234-5678-910"
        type="checkbox"
      />
      <label
        class="ToggleSwitch__SwitchLabel-sc-1n3fzwh-0 hFRJiv"
        for="test-uuid-1234-5678-910"
      >
        <span
          class="CombinedStyling__Span-sc-c3jdxk-1 ToggleSwitch__SwitchSliderSpan-sc-1n3fzwh-2 kSRMWv hLssYN"
        />
      </label>
    </div>
  </body>,
  "container": <div>
    <input
      class="ToggleSwitch__SwitchInput-sc-1n3fzwh-1 Fwlpj"
      data-testid="toggle-checkbox"
      id="test-uuid-1234-5678-910"
      type="checkbox"
    />
    <label
      class="ToggleSwitch__SwitchLabel-sc-1n3fzwh-0 hFRJiv"
      for="test-uuid-1234-5678-910"
    >
      <span
        class="CombinedStyling__Span-sc-c3jdxk-1 ToggleSwitch__SwitchSliderSpan-sc-1n3fzwh-2 kSRMWv hLssYN"
      />
    </label>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "store": {
    "@@observable": [Function],
    "dispatch": [Function],
    "getState": [Function],
    "replaceReducer": [Function],
    "subscribe": [Function],
  },
  "unmount": [Function],
}
`;

exports[`ToggleSwitch renders the switch input and label 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <input
        class="ToggleSwitch__SwitchInput-sc-1n3fzwh-1 Fwlpj"
        data-testid="toggle-checkbox"
        id="test-uuid-1234-5678-910"
        type="checkbox"
      />
      <label
        class="ToggleSwitch__SwitchLabel-sc-1n3fzwh-0 hFRJiv"
        for="test-uuid-1234-5678-910"
      >
        <span
          class="CombinedStyling__Span-sc-c3jdxk-1 ToggleSwitch__SwitchSliderSpan-sc-1n3fzwh-2 kSRMWv hLssYN"
        />
      </label>
    </div>
  </body>,
  "container": <div>
    <input
      class="ToggleSwitch__SwitchInput-sc-1n3fzwh-1 Fwlpj"
      data-testid="toggle-checkbox"
      id="test-uuid-1234-5678-910"
      type="checkbox"
    />
    <label
      class="ToggleSwitch__SwitchLabel-sc-1n3fzwh-0 hFRJiv"
      for="test-uuid-1234-5678-910"
    >
      <span
        class="CombinedStyling__Span-sc-c3jdxk-1 ToggleSwitch__SwitchSliderSpan-sc-1n3fzwh-2 kSRMWv hLssYN"
      />
    </label>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "store": {
    "@@observable": [Function],
    "dispatch": [Function],
    "getState": [Function],
    "replaceReducer": [Function],
    "subscribe": [Function],
  },
  "unmount": [Function],
}
`;

exports[`ToggleSwitch respects isChecked prop 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <input
        checked=""
        class="ToggleSwitch__SwitchInput-sc-1n3fzwh-1 Fwlpj"
        data-testid="toggle-checkbox"
        id="test-uuid-1234-5678-910"
        type="checkbox"
      />
      <label
        class="ToggleSwitch__SwitchLabel-sc-1n3fzwh-0 hFRJiv"
        for="test-uuid-1234-5678-910"
      >
        <span
          class="CombinedStyling__Span-sc-c3jdxk-1 ToggleSwitch__SwitchSliderSpan-sc-1n3fzwh-2 kSRMWv hLssYN"
        />
      </label>
    </div>
  </body>,
  "container": <div>
    <input
      checked=""
      class="ToggleSwitch__SwitchInput-sc-1n3fzwh-1 Fwlpj"
      data-testid="toggle-checkbox"
      id="test-uuid-1234-5678-910"
      type="checkbox"
    />
    <label
      class="ToggleSwitch__SwitchLabel-sc-1n3fzwh-0 hFRJiv"
      for="test-uuid-1234-5678-910"
    >
      <span
        class="CombinedStyling__Span-sc-c3jdxk-1 ToggleSwitch__SwitchSliderSpan-sc-1n3fzwh-2 kSRMWv hLssYN"
      />
    </label>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "store": {
    "@@observable": [Function],
    "dispatch": [Function],
    "getState": [Function],
    "replaceReducer": [Function],
    "subscribe": [Function],
  },
  "unmount": [Function],
}
`;

exports[`ToggleSwitch toggles checked state on click 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <input
        class="ToggleSwitch__SwitchInput-sc-1n3fzwh-1 Fwlpj"
        data-testid="toggle-checkbox"
        id="test-uuid-1234-5678-910"
        type="checkbox"
      />
      <label
        class="ToggleSwitch__SwitchLabel-sc-1n3fzwh-0 hFRJiv"
        for="test-uuid-1234-5678-910"
      >
        <span
          class="CombinedStyling__Span-sc-c3jdxk-1 ToggleSwitch__SwitchSliderSpan-sc-1n3fzwh-2 kSRMWv hLssYN"
        />
      </label>
    </div>
  </body>,
  "container": <div>
    <input
      class="ToggleSwitch__SwitchInput-sc-1n3fzwh-1 Fwlpj"
      data-testid="toggle-checkbox"
      id="test-uuid-1234-5678-910"
      type="checkbox"
    />
    <label
      class="ToggleSwitch__SwitchLabel-sc-1n3fzwh-0 hFRJiv"
      for="test-uuid-1234-5678-910"
    >
      <span
        class="CombinedStyling__Span-sc-c3jdxk-1 ToggleSwitch__SwitchSliderSpan-sc-1n3fzwh-2 kSRMWv hLssYN"
      />
    </label>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "store": {
    "@@observable": [Function],
    "dispatch": [Function],
    "getState": [Function],
    "replaceReducer": [Function],
    "subscribe": [Function],
  },
  "unmount": [Function],
}
`;
